### PR TITLE
Attempt port

### PR DESCRIPTION
**Pull request by hawken93**
This has been very demotivating to fix lol..

Lot of ugly practices relating to the database, and especially hasAnyProviderId looks very ugly in the backend and is prone to match a lot of false positives. the backend code has a commented "suggestion" to create another table instead of having a one-to-many relation inside of this table. They also have removed the dictionary in order to only match on the key, probably because it was not necessary or correct to match on the id itself. Ideally I want to change jellyfin instead because this is stupid. If there's an issue for the database work I'd like to reference it from here so they can see that this needs to be fixed :) By the way, the file in question in Jellyfin is SqliteItemRepository.cs with over 6400 lines!

There's also another error I couldn't really figure out relating to InternalId stuff in ILibraryManager which seems like this module should do something completely different instead.

Finally, there's a missing cast in MusicAlbum.cs in jellyfin which I put there instead of here, so this will error out for now but I'll link a PR to fix this in jellyfin.

### CODE REVIEW

**Comment by JustAMan** [comment]
I think @sparky8251 has the effort on sanitizing the DB going.

File: EmbyReports/Api/Activities/ReportActivitiesBuilder.cs [hunk 1]
```diff
@@ -167,7 +167,7 @@ private ReportOptions<ActivityLogEntry> GetOption(HeaderMetadata header, string
                     option.Header.CanGroup = false;
                     option.Column = (i, r) =>
                     {
-                        if (!string.IsNullOrEmpty(i.UserId))
+                        if (!i.UserId.Equals(Guid.Empty))
```
Comment (Bond-009) [inline EmbyReports/Api/Activities/ReportActivitiesBuilder.cs]: Just use `==`

File: EmbyReports/Api/Activities/ReportActivitiesBuilder.cs [hunk 2]
```diff
@@ -191,7 +191,7 @@ private ReportOptions<ActivityLogEntry> GetOption(HeaderMetadata header, string
                 case HeaderMetadata.User:
                     option.Column = (i, r) =>
                     {
-                        if (!string.IsNullOrEmpty(i.UserId))
+                        if (!i.UserId.Equals(Guid.Empty))
```
Comment (Bond-009) [inline EmbyReports/Api/Activities/ReportActivitiesBuilder.cs]: Same

File: EmbyReports/Api/ReportsService.cs [hunk 3]
```diff
@@ -456,11 +430,11 @@ private QueryResult<BaseItem> GetQueryResult(BaseReportRequest request, DtoOptio
             };
         }
 
-        protected long[] ParseIds(string value, ILibraryManager libraryManager)
+        protected Guid[] ParseIds(string value, ILibraryManager libraryManager)
         {
             if (string.IsNullOrEmpty(value))
             {
-                return Array.Empty<long>();
+                return new Guid[] { };
```
Comment (Bond-009) [inline EmbyReports/Api/ReportsService.cs]:
```suggestion
                return Array.Empty<Guid>();
```

File: EmbyReports/EmbyReports.csproj [hunk 4]
```diff
@@ -1,19 +1,21 @@
 <Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
+        <TargetFrameworks>netstandard2.0;</TargetFrameworks>
+        <AssemblyVersion>1.0.18.0</AssemblyVersion>
```
Comment (Bond-009) [inline EmbyReports/EmbyReports.csproj]: What's up with the lower version?
xml indentation is 2 spaces.

Comment (hawken93) [reply EmbyReports/EmbyReports.csproj]: Last I checked was with the big formatting fix PR where the xml initially had 4 indents, at the time I wrote this I assumed it was still like that but after poking around in jellyfin I have noticed you are going for 2 spaces. Will fix this

**Comment by hawken93** [comment]
@nvllsvm it seems like you left a review that got lost by me pushing things, was this implemented or might it have gotten lost by me force pushing?

**Comment by joshuaboniface** [comment]
@cvium Agreed, better to just revert and go from there. If we know which commit is the basis I can `git reset` and force-push it up.

File: EmbyReports/Api/ReportsService.cs [hunk 5]
```diff
@@ -456,11 +430,11 @@ private QueryResult<BaseItem> GetQueryResult(BaseReportRequest request, DtoOptio
             };
         }
 
-        protected long[] ParseIds(string value, ILibraryManager libraryManager)
+        protected Guid[] ParseIds(string value, ILibraryManager libraryManager)
```
[the remark this answers is not in the document]
Comment (hawken93) [reply EmbyReports/Api/ReportsService.cs]: Thanks, I have no idea what I'm doing and possibly a different fix would be correct here

**Comment by cvium** [comment]
@joshuaboniface The last compatible commit is 26905e18b47aa661e6b012138f198fd355ad3d52

**Comment by joshuaboniface** [comment]
Superseded by #2 - feel free to close @hawken93 if you wish, or if there's other valuable things in here please rebase your new work off the updated `master`.

**Comment by hawken93** [comment]
Thanks. I'll check if I have anything to add to #2 :)

**Pull request closed without merging**